### ISSUE TITLE
[BUG] Amethyst is always playing media when I enter my car and my phone connects to bluetooth

### ISSUE THREAD

**Issue by owen1917** [vitorpamplona/amethyst#971]
**Describe the bug**
My phone connects to my car via Bluetooth (not Android Auto - plain old Bluetooth). When I enter the car, my phone automatically connects to Bluetooth which automatically starts playing media from Amethyst, even though the app is not in active use.

**To Reproduce**
1. Connect your Android device to a car media system via Bluetooth
2. Leave the car/disconnect from the car Bluetooth
4. Open Amethyst, casually scroll your feed (making sure to scroll past some video content)
5. Put the Phone to sleep
6. Enter the car/connect to the car Bluetooth
7. Car media system is playing the audio from the last video that I viewed on my Amethyst feed

**Expected behavior**
Amethyst should ideally never be playing media through the car sound system, unless I explicitly open the app and watch a video with the app in the foreground.

**Video and Screenshots**
![IMG_20240702_082353_725](https://github.com/vitorpamplona/amethyst/assets/123519473/e23f4583-c67c-4008-97de-50ab315c24e9)

![Screenshot_20240702-082419](https://github.com/vitorpamplona/amethyst/assets/123519473/fef623cd-8bc7-4cc8-aee2-942356a43f90)

**Device (please complete the following information):**
 - Pixel 6a
 - Android 14 / GrapheneOS 
 - App Version 0.88.7-PLAY

**Bounty (in Bitcoin sats) offered for a solution**
20k sats

**Comment by davotoula** [comment]
Oh yeah, I get exactly the same thing.

It could be an android "feature" rather than an amethyst bug but let's see what others think.

I tried a few things. Android keeps the latest used app in the "media drawer".

Sometimes the media keeps playing even after closing amethyst.. Could be background play functionality but there is no sound from phone. 

This will then start playing sound on a new Bluetooth connection like a car.

**Comment by owen1917** [comment]
> Oh yeah, I get exactly the same thing.
> 
> It could be an android "feature" rather than an amethyst bug but let's see what others think.
> 
> I tried a few things. Android keeps the latest used app in the "media drawer".
> 
> Sometimes the media keeps playing even after closing amethyst.. Could be background play functionality but there is no sound from phone.
> 
> This will then start playing sound on a new Bluetooth connection like a car.

Yes, I believe you're right, this is probably Android functionality at the core. 

I believe expected behaviour is that the player disappears when the app goes to background. This works as expected. 

In reality though, as a user I often use Amethyst and put my device to sleep. When I get into my car Im not actively using Amethyst so despite it being the last app used I wouldn't expect it to start playing.

In comparison, Primal doesn't appear to use this functionality at all so it doesn't suffer the same problem.

I wonder if there is any use case for Amethyst using this Android feature at all?

**Comment by davotoula** [comment]
![17201315489060.png](https://github.com/vitorpamplona/amethyst/assets/1747287/602afadc-45d0-425a-80a6-7f61c80a44ca)

Try disabling Now playing notifications?

**Comment by owen1917** [comment]
> Try disabling Now playing notifications?

I tried this, unfortunately the issue still occurs

**Comment by greenart7c3** [comment]
@vitorpamplona I think I managed to reproduce this but I don't have a Bluetooth device to try to fix the issue.
It happens when the Bluetooth is off and you turn it on and connect to a device.
When you do that the video notification starts playing again

**Comment by vcavallo** [comment]
I know this may not be ideal, but if you _see_ the video notification bubble, you can tap and hold on it to dismiss it. 

![Screenshot_20241026-170243.png](https://github.com/user-attachments/assets/2c88ff09-a808-4387-94de-433390c67212)


Obviously not helpful if the notification isn't visible first though. And still annoying to have to remember to wipe it.

**Comment by KotlinGeekDev** [comment]
I wonder if this issue can be closed.

**Comment by greenart7c3** [comment]
It still happens to me sometimes, haven't tried to debug it to see why this is happening

**Comment by davotoula** [comment]
I think the current playing amethyst Media is forced to now playing

**Comment by KotlinGeekDev** [comment]
Interesting...

**Comment by caydyan** [comment]
Opened PR https://github.com/vitorpamplona/amethyst/pull/3213 for this.

The patch blocks Bluetooth/headset media-button resume events while Amethyst is backgrounded for normal feed video sessions, while still allowing the explicit PiP/background media session to handle controls.

Validation:
- `git diff --check`
- Attempted the focused Gradle unit test, but local Gradle dependency resolution failed before compilation due Maven TLS handshake errors.

BTC bounty address: `bc1qev5ant33v5y89qqjvcf4mh9hlax5svqf5xd7gc`

**Comment by Christab30** [comment]
Research summary before code: #971 is still open and users have confirmed the Bluetooth/media-session resume behavior persists. I found closed PR #3213, which tried to solve this by tracking foreground state and consuming PLAY/PLAY_PAUSE/HEADSETHOOK media-button events for normal feed sessions while preserving explicit `BackgroundMedia`/PiP playback. That PR was closed unmerged and without review comments, so I will not simply resubmit it.

I’m checking the current `PlaybackService`/`MediaSessionPool` lifecycle first and intend to keep the fix Android-only, narrowly scoped to preventing external media-button resume while the app is backgrounded, with an explicit exception for user-requested background/PiP playback. I’ll add a regression test for the decision logic and will only open the PR with current Play/F-Droid build/test evidence.

@vitorpamplona, if #971 is still wanted, is that behavior/scope acceptable on current main? @ghost, is the original 20k-sat bounty still active for an accepted and merged fix?
